include/exec/task.hpp

```diff
@@ -267,6 +374,30 @@ namespace exec {
           this->__data_.template emplace<2>(std::current_exception());
         }
 
+        template <stdexec::sender _Awaitable>
+          requires(
+            !stdexec::__decays_to<_Awaitable, complete_inline_t>
+            && __completion_scheduler_provider<_Context>)
+        decltype(auto) await_transform(_Awaitable&& __awaitable) noexcept {
+          return stdexec::as_awaitable(
+            stdexec::transfer(
+              (_Awaitable&&) __awaitable,
+              stdexec::get_completion_scheduler<stdexec::set_value_t>(__context_)),
+            *this);
+        }
+
+        template <stdexec::__decays_to<complete_inline_t> _Awaitable>
+        decltype(auto) await_transform(_Awaitable&& __awaitable) noexcept {
+          __context_.set_scheduler(__awaitable.__scheduler_);
+          return stdexec::as_awaitable(stdexec::schedule(__awaitable.__scheduler_), *this);
+        }
```

Comment:
I need to investigate how this works. I was imagining that complete_inline would be a sender adaptor.

Reply:
Currently, it's a special awaitable just to set the scheduler from within a coro. Currently it's the only way to modify the current scheduler. I plan to add a specialized on(sched, task) that sets the initial scheduler, which otherwise defaults to inline scheduler unless it is suspended as an awaitable in a parent context where inherits the current scheduler of its parent